src/runtime/classbase.cs

```diff
@@ -233,7 +233,86 @@ public static IntPtr tp_str(IntPtr ob)
             }
             try
             {
+                Console.WriteLine("classbase tp_str");
+                //As per python doc:
+                //The return value must be a string object.  If a class defines __repr__() but not __str__(),
+                //then __repr__() is also used when an “informal” string representation of instances of that
+                //class is required.
+                //In C#, everything provides ToString(), so the check here will be whether the type explicitly
+                //provides ToString() or if it is language provided (i.e. the fully qualified type name as a string)
+
+                //First check which type in the object hierarchy provides ToString()
+                //ToString has two "official" overloads so loop over GetMethods to get the one without parameters
+                var instType = co.inst.GetType();
+                foreach (var method in instType.GetMethods())
+                {
+                    Console.WriteLine("classbase tp_str in loop");
+                    //TODO this could probably be done more cleanly with Linq
+                    if (!method.IsPublic) continue; //skip private/protected methods
+                    if (method.Name != "ToString") continue; //only look for ToString
+                    if (method.DeclaringType == typeof(object)) continue; //ignore default from object
+                    if (method.GetParameters().Length != 0) continue; //ignore Formatter overload of ToString
```

Comment:
Can't you express all this in a straight `GetMethod` call?

Reply:
@filmor I don't doubt it, I just don't know the syntax offhand

Reply:
@filmor I'll give this a shot